applications/launchpad/docker_rig/promtail.config.yml

```diff
@@ -0,0 +1,38 @@
+server:
+  http_listen_port: 9080
+  grpc_listen_port: 0
+
+positions:
+  filename: /tmp/positions.yaml # This location needs to be writeable by Promtail.
+
+client:
+  url: http://loki:3100/loki/api/v1/push
+
+scrape_configs:
+ - job_name: system
+   pipeline_stages:
+   - docker:
+   - regex:
+      # version source: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string with some modification
+      expression: ".*[^\\d\\.\\/](?P<version>(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?)[^\\.\\d:]"
+   - regex:
+      source: filename
+      # network, service
+      expression: "^\\/var\\/log\\/(?P<network>.+?)\\/(?P<service>.+?)\\/.+$"
+   - regex:
+      # time, source, level, content
+      expression: "^(?P<time>\\d{4}-\\d{2}-\\d{2}\\s\\d{2}:\\d{2}:\\d{2}\\.\\d{9})\\s\\[(?P<source>(\\w|:)+)\\]\\s(?P<level>\\w+)\\s+(?P<content>.+)$"
+   - labels:
+      service:
+      source:
+      level:
+      version:
+      network:
+   - timestamp:
+      source: time
+      format: RFC3339Nano
+   - output:
+      source: content
+   static_configs:
+   - labels:
+      __path__: /var/log/**/*.log  # The path matching uses a third party library: https://github.com/bmatcuk/doublestar
```

Comment:
this needs to change depending on OS, not sure how (?)
possibly we could have files per OS and point to that with docker command (see docker-compose)